src/NServiceBus.Gateway/Routing/RouteToGatewayBehaviour.cs

```diff
@@ -1,23 +1,44 @@
 ﻿namespace NServiceBus
 {
     using System;
+    using System.Collections.Generic;
     using System.Threading.Tasks;
     using Gateway.Routing;
     using Pipeline;
     using Routing;
 
     class RouteToGatewayBehavior : Behavior<IRoutingContext>
     {
-        public RouteToGatewayBehavior(string gatewayAddress)
+        public RouteToGatewayBehavior(string gatewayAddress, IList<string> configuredSiteKeys)
         {
             this.gatewayAddress = gatewayAddress;
+            this.configuredSiteKeys = configuredSiteKeys;
         }
 
         public override Task Invoke(IRoutingContext context, Func<Task> next)
         {
-
             if (context.Extensions.TryGet(out RouteThroughGateway _))
             {
+                if (context.Message.Headers.TryGetValue(Headers.DestinationSites, out var siteKeyHeader))
+                {
+                    var siteKeys = siteKeyHeader.Split(',');
+
+                    var unknownSites = new List<string>();
+
+                    foreach (var siteKey in siteKeys)
+                    {
+                        if (!configuredSiteKeys.Contains(siteKey))
+                        {
+                            unknownSites.Add(siteKey);
+                        }
+                    }
+
+                    if (unknownSites.Count > 0)
+                    {
+                        throw new Exception($"Sites with keys `{string.Join(",", unknownSites)}` was not found in the list of configured sites. Please make sure to configure it or remove it from the call to `{nameof(SendOptionsExtensions.RouteToSites)}`");
```

Comment:
This exception message needs some work. Are "keys" even something someone using the Gateway would know about?

They're also much more likely to have used `SendToSites`, so talking about `SendOptionsExtensions.RouteToSites` here seems like it would be confusing. Do we even need that sentence?

Also, there's a grammar problem. Since "Sites" is plural, it should be "were" and not "was".

So, maybe something like:

"The following sites have not been configured: `{string.Join(",", unknownSites)}`"